### PR TITLE
Consolidate Quickstart CI script

### PR DESCRIPTION
**Pull request by jrcrespoh**

### CODE REVIEW

**Comment by jrcrespoh** [comment]
At this point, the remaining failures are due to the quickstarts failing to pass their unit or UI tests.

Quickstarts needing test fixing:
 - [ ] AdMob
 - [ ] Authentication
 - [ ] Database
 - [ ] Firestore
 - [ ] Messaging

**Comment by jrcrespoh** [comment]
> At this point, the remaining failures are due to the quickstarts failing to pass their unit or UI tests.
> 
> Quickstarts needing test fixing:
> 
> * [ ]  AdMob
> * [ ]  Authentication
> * [ ]  Database (cocoapods)
> * [ ]  Firestore
> * [ ]  Messaging

Green CI achieved by disabling testing for the above quickstarts; all other quickstarts both build and test successfully. The following spm quickstarts now run on GHA macOS 11:

- [x] A/B Testing
- [x] Crashlytics
- [x] Database
- [x] Performance

File: scripts/test.sh
Comment (jrcrespoh) [inline scripts/test.sh:85]: `DESTINATION` is not used when targeting watchOS: previously there was trouble getting `xcodebuild` to use the right simulator / device, and without a `-destination` flag it works fine for the moment.

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/README.md
Comment (paulb777) [inline performance/SwiftUIPerformanceQuickstart/PerformanceExample/README.md:12]: The saliency changes should be in a different PR

File: scripts/test.sh
Comment (paulb777) [inline scripts/test.sh:1]: Nice consolidation of the scripts!

**Review by paulb777 — COMMENTED** [review]
Please make sure UITests are still running. For example, I don't see them in https://github.com/firebase/quickstart-ios/runs/3390100787?check_suite_focus=true

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/README.md
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/README.md:12]: They were merged with https://github.com/firebase/quickstart-ios/pull/1257. I needed to merge those changes in to fix performance CI (storage rules only work for saliency changes). What should I do on this PR?

**Comment by jrcrespoh** [comment]
> Please make sure UITests are still running. For example, I don't see them in https://github.com/firebase/quickstart-ios/runs/3390100787?check_suite_focus=true

@paulb777 Specific tests were disabled on purpose since they weren't passing: https://github.com/firebase/quickstart-ios/pull/1258#issuecomment-903069392

**Comment by paulb777** [comment]
Were the tests disabled previously as well?

**Comment by jrcrespoh** [comment]
> Were the tests disabled previously as well?

They weren't disabled on purpose. Secrets were not being properly propagated (see https://github.com/firebase/quickstart-ios/pull/1258/commits/4f5d4a96f906bc64abb189635540f17472759cb9), so the previous script simply skipped testing.

**Review by paulb777 — APPROVED** [review]
This is a great simplification/reorganization. Thanks @jrcrespoh !

I'm good to merge assuming that there is no regression in which builds and tests run.

**Review by morganchen12 — APPROVED** [review]
LGTM, but please wait for Paul/Gran approval before merging.

**Review by granluo — APPROVED** [review]
Thanks for updating the scripts. This does simplify the workflows and increase the maintainability.

**Comment by ncooke3** [comment]
Thanks @jrcrespoh! This is terrific! 🎉

**Pull request merged**
